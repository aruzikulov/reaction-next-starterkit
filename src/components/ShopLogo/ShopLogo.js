import React, { Component } from "react";
import PropTypes from "prop-types";
import Link from "components/Link";

export default class ShopLogo extends Component {
  static propTypes = {
    /**
     * The primary shop's logo url
     */
    shopLogoImgUrl: PropTypes.string,
    /**
     * The primary shop's name
     */
    shopName: PropTypes.string.isRequired
  }

  render() {
    const { shopLogoImgUrl, shopName } = this.props;

    return (
      <Link route="/">
        {
          shopLogoImgUrl ? (
            <img src={shopLogoImgUrl} alt={shopName} />
          ) : (
            shopName
          )
        }
      </Link>
    );
  }
}

