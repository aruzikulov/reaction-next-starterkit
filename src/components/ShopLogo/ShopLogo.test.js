import React from "react";
import renderer from "react-test-renderer";
import ShopLogo from "./ShopLogo";

test("Should render shop's logo image", () => {
  const component = renderer.create((
    <ShopLogo shopLogoImgUrl="logo.png" shopName="Reaction" />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should render the shop's name", () => {
  const component = renderer.create((
    <ShopLogo shopName="Reaction" />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
