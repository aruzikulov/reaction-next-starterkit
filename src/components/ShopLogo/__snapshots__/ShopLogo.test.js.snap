// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render shop's logo image 1`] = `
<a
  className="WithTracking-Link--link-1"
  href="/"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
  tracking={
    Object {
      "getTrackingData": [Function],
      "trackEvent": [Function],
    }
  }
>
  <img
    alt="Reaction"
    src="logo.png"
  />
</a>
`;

exports[`Should render the shop's name 1`] = `
<a
  className="WithTracking-Link--link-1"
  href="/"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
  tracking={
    Object {
      "getTrackingData": [Function],
      "trackEvent": [Function],
    }
  }
>
  Reaction
</a>
`;
